TODO

```diff
@@ -29,8 +26,34 @@ Janitorial Clean-ups:
 
 Features:
 
-* replace all canonicalize_file_name() invocations by chase_symlinks(), in
-  particulr those where a rootdir is relevant.
+* set ProtectSystem=strict for all our usual services.
+
+* maybe add gpt-partition-based user management: each user gets his own
+  LUKS-encrypted GPT partition with a new GPT type. A small nss module
+  enumerates users via udev partition enumeration. UIDs are assigned in a fixed
+  way: the partition index is added as offset to some fixed base uid. User name
+  is stored in GPT partition name. A PAM module authenticates the user via the
+  LUKS partition password. Benefits: strong per-user security, compatibility
+  with stateless/read-only/verity-enabled root.
+
+* gpt-auto logic: introduce support for discovering /var matching an image. For
+  that, use a partition type UUID that is hashed from the OS name (as encoded
+  in /etc/os-release), the architecture, and 4 new bits from the gpt flags
+  field of the root partition. This way can easily support multiple OS
+  installations on the same GPT partition table, without problems with
+  unmatched /var partitions.
+
+* gpt-auto logic: related to the above, maybe support a "secondary" root
+  partition, that is mounted to / and is writable, and where the actual root's
+  /usr is mounted into.
+
+* add dm-verity boots, and in nspawn (libcryptsetup knows this, should be relatively straight-forward)
+
+* machined: add apis to query /etc/machine-info data of a container
+
+* .mount and .swap units: add Format=yes|no option that formats the partition before mounting/enabling it, implicitly
```

Comment:
my intention with this was to use the "mkfs" wrapper that util-linux offers (which does what the "fsck" wrapper does for hte various per-filesystem fscks, but for mkfs), so that we don't have to know anything about file systems really, and only support options that are generic enough that most backends speak them. That probably boils down to supporting precisely "-U" (specify uuid) and "-L" (specify label), and nothing else. But I think that's fine: we'd only support the file system's default, and people who want something else can call the tools directly.

The "tmp" crypttab option suggests that in many cases just using the defaults is entirely sufficient for what people need. And my idea was mostly to generalize that concept, because I think it shouldn't be a crypttab option at all, but an fstab option. Consider:

```
/dev/sda7 /var/tmp ext4 x-systemd.format 0 0
```

And that would just mean that systemd would invoke "/usr/bin/mkfs -t ext4 /dev/sda7" right before invoking "/usr/bin/mount -t ext4 /dev/sda7 /var/tmp"...

You get the idea...